src/Streamly/Internal/Data/Array/Stream/Foreign.hs

```diff
@@ -457,18 +525,23 @@ _toArraysOf n = FL.chunksOf n (A.writeNF n) FL.toStream
 -}
 
 -------------------------------------------------------------------------------
--- Parsing
+-- Elimination - running element parsers
 -------------------------------------------------------------------------------
 
 -- GHC parser does not accept {-# ANN type [] NoSpecConstr #-}, so we need
 -- to make a newtype.
 {-# ANN type List NoSpecConstr #-}
 newtype List a = List {getList :: [a]}
 
+-- This can be generalized to any type provided it can be unfolded to a stream
+-- and it can be combined using a semigroup operation.
+--
+-- XXX This should be written using CPS (as parseK) if we want it to scale wrt
+-- to the number of times it can be called on the same stream.
```

Comment:
Can you open an issue and assign it to me?

Reply:
ok.